components/centraldashboard/public/components/main-page.js

```diff
@@ -173,10 +173,12 @@ export class MainPage extends utilitiesMixin(PolymerElement) {
      * @param {Event} ev AJAX-response
      */
     _onHasWorkgroupResponse(ev) {
-        const {user, hasWorkgroup, hasAuth} = ev.detail.response;
+        const {user, hasAuth, namespaceCreation} = ev.detail.response;
         this._setIsolationMode(hasAuth ? 'multi-user' : 'single-user');
-        if (hasAuth && !hasWorkgroup) {
-            this.user = user;
+        this.user = user;
+        if (!namespaceCreation) {
+            this._setRegistrationFlow(false);
```

Comment:
Same logic would need to be in place [here](https://github.com/kubeflow/kubeflow/blob/253890cb13aef2da83468d0ab6d244a9bff26144/components/centraldashboard/public/components/main-page.js#L324-L329) as well

Reply:
@lalithvaka Since it's defaulted to false here, I don't think we don't need to change it. If the deployment is for a single user, they will need to have a namespace.

Reply:
Got it. Thank you.

Reply:
Please also update the tests for this flow